x-pack/plugins/cloud_integrations/cloud_experiments/common/types.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { FEATURE_FLAG_NAMES, METRIC_NAMES } from './constants';
+
+/**
+ * The contract of the setup lifecycle method.
+ */
+export interface CloudExperimentsPluginSetup {
+  /**
+   * Identifies the user in the A/B testing service.
+   * For now, we only rely on the user ID. In the future, we may request further details for more targeted experiments.
+   * @param userId The unique identifier of the user in the experiment.
+   * @param userMetadata Additional attributes to the user.
```

Comment:
Good point! Added. I also added the `@deprecated` flag to this API because we are planning to swap the dependencies once we _clean up_ the `cloud` plugin's dependency tree.